state/watcher.go

```diff
@@ -505,6 +505,20 @@ func (st *State) WatchModelMachines() StringsWatcher {
 	return newLifecycleWatcher(st, machinesC, members, filter, nil)
 }
 
+// WatchModelMachinesLXDProfile returns a StringsWatcher that notifies
+// when a charm upgrade requests a profile upgrade too.
+func (st *State) WatchModelMachinesLXDProfile() StringsWatcher {
```

Comment:
Not doing what I want currently.  If the UpgradeLXDProfileApplication and UpgradeLXDProfileCharmURL is set, trigger the watcher.  However it should be changed.